web/packages/teleport/src/lib/term/tty.ts

```diff
@@ -168,13 +197,46 @@ class Tty extends EventEmitterWebAuthnSender {
 
   _processAuditPayload(payload) {
     const event = JSON.parse(payload);
+    // received a new/updated file transfer request
+    if (event.event === EventType.FILE_TRANSFER_REQUEST) {
+      this.emit(EventType.FILE_TRANSFER_REQUEST, event);
+    }
+
+    // received a file transfer approval
+    if (event.event === EventType.FILE_TRANSFER_REQUEST_APPROVE) {
+      const isDownload = event.download === true;
+      let pendingFile: File = null;
+      // if the approval is for an upload, fetch the file pending upload
+      if (!isDownload) {
+        const locationAndName = event.location + event.filename;
+        pendingFile = this._getPendingFile(locationAndName);
+        // cleanup if file exists. It's ok if it doesn't exist, we check thaat in the handler
+        if (pendingFile) {
+          delete this._pendingUploads[locationAndName];
```

Comment:
Yeah I guess that would happen with this code but it is a somewhat unrealistic case that you wouldn't want to upload the same file to the same place at the same time right? A way to prevent this would be an error earlier up but because its such an edge case, an error at any spot would probably suffice. Let me know your thoughts if you have any more!

Could always do something like, set the file initially the way we are now with locationAndName and then when we receive the file transfer request, we can update the key with the ID? idk if all that would be worth it or make sense, just spitballin.